### PR TITLE
[4.0] Filtered item information [a11y]

### PR DESCRIPTION
**Pull request by brianteeman**
In a previous pull request I added code which adds a caption to every table AND indicates how the table is sorted. This is only visible to screen readers. This is very important for accessibility so a user will now all the relevant information for the table being displayed

To improve this further it should also include how the table is filtered.

I have only enabled this for the table of contacts at this time. If/when approved I will do the same for the other tables.

This PR will also allow the designers to add a visual display of the filters which people like @angieradtke have suggested

I also had to use real label strings for the filters instead of re-using the string from the option. I should have done this before but ...

Thanks to @wilsonge for fixing the js

(as this is javascript it will require either` npm i ` or `node build.js --compile-js` to test it)

Testing is really simple

Once the pr has been applied and the js compiled go to the list of all contacts in the admin

If you view source you will see
image

Play around adding filters and it will change as below
image

A screen reader will see the text as
`Table of Contacts, Sorted by: Title - descending, Filtered by: Status: Published`

### CODE REVIEW

**Review by wilsonge — APPROVED** [review]

**Comment by Quy** [comment]
Add a space after `Sorted by:`

File: build/media_source/system/js/searchtools.es6.js [hunk 1]
```diff
@@ -306,6 +312,24 @@ Joomla = window.Joomla || {};
...
+
+        if (element.multiple === true) {
+          const selectedOptions = element.querySelectorAll('option:checked');
+          const selectedTextValues = Array.from(selectedOptions).map((el) => el.text);
+          captionContent = `${element.labels[0].textContent} - ${selectedTextValues.join()}`;
+        } else {
+          captionContent = `${element.labels[0].textContent} - ${element.options[element.selectedIndex].text}`;
+        }
+
+        document.getElementById('filteredBy').textContent += captionContent;
```
Comment (C-Lodder) [inline build/media_source/system/js/searchtools.es6.js]: You can replace `document.getElementById('filteredBy')` with `filteredByCaption`, as you've defined it as a variable above

File: build/media_source/system/js/searchtools.es6.js [hunk 2]
```diff
@@ -306,6 +312,24 @@ Joomla = window.Joomla || {};
...
+      // Add all active filters to the table caption for screen-readers
+      const filteredByCaption = document.getElementById('filteredBy');
+
+      // The caption won't exist if no items match the filters so check for the element first
+      if (filteredByCaption) {
+        let captionContent = '';
+
+        if (element.multiple === true) {
+          const selectedOptions = element.querySelectorAll('option:checked');
+          const selectedTextValues = Array.from(selectedOptions).map((el) => el.text);
```
Comment (C-Lodder) [inline build/media_source/system/js/searchtools.es6.js]: You can't use `Array.from(...)` as it's not support in IE11. Use `[].slice.call(selectedOptions)`

File: build/media_source/system/js/searchtools.es6.js (hunk 1 above)
Comment (brianteeman) [reply build/media_source/system/js/searchtools.es6.js]: Done

File: build/media_source/system/js/searchtools.es6.js (hunk 2 above)
Comment (brianteeman) [reply build/media_source/system/js/searchtools.es6.js]: /me confused

Comment (C-Lodder) [reply build/media_source/system/js/searchtools.es6.js]: about?

Comment (brianteeman) [reply build/media_source/system/js/searchtools.es6.js]: are we supporting ie11 or not

Comment (C-Lodder) [reply build/media_source/system/js/searchtools.es6.js]: In the JS, yes apparently

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Right I guess I need to do all the other components now

**Pull request merged**